### PR TITLE
*: add encode SQL digest function

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43193

Problem Summary:
Add encode SQL digest function.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/cluster_table_test.go [hunk 1]
```diff
@@ -392,6 +392,32 @@ func TestFunctionDecodeSQLDigestsPrivilege(t *testing.T) {
...
+	digest2 := parser.DigestHash(q2)
+	q3 := "select id, v from test_func_decode_sql_digests where id = 1 for update"
+	digest3 := parser.DigestHash(q3)
+
+	tk.MustQuery(fmt.Sprintf("select tidb_encode_sql_digest(\"%s\")", q1)).Check(testkit.Rows(digest1.String()))
+	tk.MustQuery(fmt.Sprintf("select tidb_encode_sql_digest(\"%s\")", q2)).Check(testkit.Rows(digest2.String()))
+	tk.MustQuery(fmt.Sprintf("select tidb_encode_sql_digest(\"%s\")", q3)).Check(testkit.Rows(digest3.String()))
+
+	tk.MustQuery("select tidb_decode_sql_digests(null)").Check(testkit.Rows("<nil>"))
+	tk.MustGetErrCode("select tidb_decode_sql_digests()", 1582)
```
Comment (kennytm) [inline executor/cluster_table_test.go:418]: why are these testing `tidb_decode_sql_digests` :thinking:

Comment (jackysp) [reply executor/cluster_table_test.go:418]: I copied it from the latest case. Addressed, PTAL.

**Comment by jackysp** [comment]
PTAL @kennytm

**Comment by jackysp** [comment]
PTAL @AndreMouche

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**